Wikipedia/UI-V5/WMFRelatedTitleListDataSource.h

```diff
@@ -0,0 +1,37 @@
+//
+//  WMFRelatedTitleListDataSource.h
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/4/15.
+//  Copyright (c) 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <SSDataSources/SSArrayDataSource.h>
+#import "WMFArticleListDataSource.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class WMFRelatedSearchFetcher;
+@class MWKSavedPageList;
+@class MWKDataStore;
+@class MWKTitle;
+
+@interface WMFRelatedTitleListDataSource : SSArrayDataSource
+    <WMFArticleListDataSource>
+
+
+- (instancetype)initWithTitle:(MWKTitle*)title
+                    dataStore:(MWKDataStore*)dataStore
+                savedPageList:(MWKSavedPageList*)savedPageList
+    numberOfExtractCharacters:(NSUInteger)numberOfExtractCharacters;
+
+- (instancetype)initWithTitle:(MWKTitle*)title
+                    dataStore:(MWKDataStore*)dataStore
+                savedPageList:(MWKSavedPageList*)savedPageList
+    numberOfExtractCharacters:(NSUInteger)numberOfExtractCharacters
+                      fetcher:(WMFRelatedSearchFetcher*)fetcher NS_DESIGNATED_INITIALIZER;
```

Comment:
@coreyfloyd IMO having designated & convenience initializers is pretty standard, and if we required separate headers for all designated initializers it would be a non-trivial amount of tedious work.  let's discuss on hangout/IRC